dataprofiler/profilers/profile_builder.py

```diff
@@ -603,8 +603,11 @@ def clean_data_and_get_base_stats(
             sample_ind_generator.close()
 
         # If min_true_samples exists, sort
-        if min_true_samples > 0 or sample_ids is None:
-            true_sample_list = sorted(true_sample_list)  # type: ignore[assignment]
+        true_sample_list = (
+            sorted(true_sample_set)
+            if min_true_samples > 0 or sample_ids is None
+            else list(true_sample_set)
+        )
```

Comment:
Definitely agree with the renaming / typing fix as a result. My only concern is the effect on memory wrt large data if `true_sample_list` is large. Thoughts?

Reply:
Resolving this and continuing thread in other comment.